### PR TITLE
Component GET endpoint

### PR DESCRIPTION
**Pull request by axelri**
Add endpoint to fetch all workflows belonging to a certain component

### CODE REVIEW

**Comment by axelri** [comment]
PTAL @honnix @danielnorberg

File: styx-api-service/src/main/java/com/spotify/styx/api/WorkflowResource.java [hunk 1]
```diff
@@ -97,6 +101,14 @@ public WorkflowResource(Storage storage, DockerImageValidator dockerImageValidat
     );
   }
 
+  private Response<List<Workflow>> component(String cid) {
```
Comment (honnix) [inline styx-api-service/src/main/java/com/spotify/styx/api/WorkflowResource.java]: How about name it as `workflows()`

File: styx-common/src/main/java/com/spotify/styx/storage/AggregateStorage.java [hunk 2]
```diff
@@ -158,6 +158,11 @@ public void storeWorkflow(Workflow workflow) throws IOException {
   }
 
   @Override
+  public List<Workflow> component(String componentId) throws IOException {
```
Comment (honnix) [inline styx-common/src/main/java/com/spotify/styx/storage/AggregateStorage.java]: same here, name it as `workflows`?

File: styx-common/src/main/java/com/spotify/styx/storage/DatastoreStorage.java [hunk 3]
```diff
@@ -327,6 +327,33 @@ public void updateNextNaturalTrigger(WorkflowId workflowId, Instant instant) thr
     return map;
   }
 
+  public List<Workflow> componentWorkflows(String componentId) throws IOException {
```
Comment (honnix) [inline styx-common/src/main/java/com/spotify/styx/storage/DatastoreStorage.java]: same here, name it as `workflows`?

File: styx-common/src/main/java/com/spotify/styx/storage/DatastoreStorage.java [hunk 4]
```diff
@@ -327,6 +327,33 @@ public void updateNextNaturalTrigger(WorkflowId workflowId, Instant instant) thr
...
+    final Key componentKey = componentKeyFactory.newKey(componentId);
+
+    final List<Workflow> workflows = Lists.newArrayList();
+    final EntityQuery query = Query.newEntityQueryBuilder()
+        .setKind(KIND_WORKFLOW)
+        .setFilter(PropertyFilter.hasAncestor(componentKey))
```
Comment (honnix) [inline styx-common/src/main/java/com/spotify/styx/storage/DatastoreStorage.java]: 👍

File: styx-common/src/main/java/com/spotify/styx/storage/InMemStorage.java [hunk 5]
```diff
@@ -118,6 +118,13 @@ public void storeWorkflow(Workflow workflow) throws IOException {
   }
 
   @Override
+  public List<Workflow> component(String componentId) throws IOException {
```
Comment (honnix) [inline styx-common/src/main/java/com/spotify/styx/storage/InMemStorage.java]: same here, name it as `workflows`?

File: styx-common/src/main/java/com/spotify/styx/storage/Storage.java [hunk 6]
```diff
@@ -114,6 +114,14 @@
   Optional<Workflow> workflow(WorkflowId workflowId) throws IOException;
 
   /**
+   * Get a list of {@link Workflow} definitions belonging to a component
+   *
+   * @param componentId The workflow to get
+   * @return A list of {@link Workflow} definitions
+   */
+  List<Workflow> component(String componentId) throws IOException;
```
Comment (honnix) [inline styx-common/src/main/java/com/spotify/styx/storage/Storage.java]: same here, name it as `workflows`?

File: styx-api-service/src/main/java/com/spotify/styx/api/WorkflowResource.java [hunk 7]
```diff
@@ -97,6 +101,14 @@ public WorkflowResource(Storage storage, DockerImageValidator dockerImageValidat
...
+    try {
+      return Response.forPayload(storage.component(cid));
+    } catch (IOException e) {
+      return Response.forStatus(Status.INTERNAL_SERVER_ERROR.withReasonPhrase("Failed to get workflows"));
```
Comment (honnix) [inline styx-api-service/src/main/java/com/spotify/styx/api/WorkflowResource.java]: Let's make it clear, like "Failed to get workflows of component <component_id>"

File: styx-api-service/src/main/java/com/spotify/styx/api/WorkflowResource.java (hunk 1 above)
Comment (honnix) [inline styx-api-service/src/main/java/com/spotify/styx/api/WorkflowResource.java]: Also, let's call it `componentId` to be more explicit.

**Comment by honnix** [comment]
LGTM

**Pull request merged**